### DISCUSSION TITLE
GitHub GraphQL datasource: skipping empty item

### DISCUSSION THREAD

**Discussion by mjnowen** [renovatebot/renovate#30590] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

gitlab.com/renovate-bot/renovate-runner / latest

### Please tell us more about your question or problem

Hi,

I'm wondering if someone could help me understand this INFO log message I get when Renovate runs?

I have this declared in my golang project -> "go.mod" file:

```
require github.com/coreos/go-systemd/v22 v22.5.0
```

I don't understand what the log message below means or how to fix it, as everything seems to be fine.

```
 INFO: GitHub GraphQL datasource: skipping empty item (repository=<redacted>)
       "packageName": "coreos/go-systemd",
       "baseUrl": "https://api.github.com/"
```

I do get some other references to "graphql" in my renovate log (LOG_LEVEL=debug), but they seem **unrelated** and just simple API rate limiting which Renovate is handling for me automatically:

```
DEBUG: POST https://api.github.com/graphql = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=403 retryCount=0, duration=26) (repository=<redacted>)
DEBUG: Retry-After: will retry https://api.github.com/graphql after 60 seconds (repository=<redacted>)
DEBUG: POST https://api.github.com/graphql = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=403 retryCount=0, duration=34) (repository=<redacted>)
DEBUG: Retry-After: will retry https://api.github.com/graphql after 60 seconds (repository=<redacted>)
DEBUG: POST https://api.github.com/graphql = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=403 retryCount=0, duration=18) (repository=<redacted>)
DEBUG: Retry-After: will retry https://api.github.com/graphql after 60 seconds (repository=<redacted>)
```

Thank you,
Mike

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by mjnowen** [comment]
Any ideas/suggestions on what this log message means? thx

**Reply by HonkingGoose** [reply]
You're self-hosting on GitLab. But Renovate is fetching things from the GitHub API:

```
 INFO: GitHub GraphQL datasource: skipping empty item (repository=<redacted>)
       "packageName": "coreos/go-systemd",
       "baseUrl": "https://api.github.com/"
```

_Maybe_ Renovate tries to fetch the changelogs for your dependencies from GitHub, and fails, because of a missing GitHub token? Please read the [Renovate docs, Running Renovate on a non-GitHub platform](https://docs.renovatebot.com/key-concepts/changelogs/#running-renovate-on-a-non-github-platform) for more info.

Or I'm totally wrong about the above, and you have some other problem. In that case, you'll have to wait a bit for a maintainer to respond. 😉

**Reply by mjnowen** [reply]
Thanks for the helpful notes. I can see from my renovate DEBUG log that I am successfully applying my `GITHUB_COM_TOKEN`. I can also see the odd, few api throttle timeouts when I do get throttled and Renovate successfully backs off and is successful on subsequent retry.

If I was to guess, I *think* Renovate is hitting an issue with parsing the semVer of this specific go package: `https://github.com/coreos/go-systemd/releases` ie: `go-systemd v22.5.0` with a TAG value of: `v22.5.0` is causing it to report: "skipping empty item" <--which is the bit I'm confused by. AS this *should* be working. So, I'm guessing there is something specific to this go repo which ideally someone at Renovate could help advise what needs fixing here?

SemVer was only adopted by this package on version: `v22.0.0` so maybe, Renovate is tripping on the earlier versions (which I don't care about).

https://github.com/coreos/go-systemd/releases/tag/v22.0.0
https://github.com/coreos/go-systemd/releases/tag/v21
